File: python/hardware/cooling/air_cooling.py
from datetime import date
from hardware.cooling.cooling import Cooling

class AirCooling(Cooling):
    def __init__(self, serial_number="", installation_date=None, physically_installed=False, max_thermal_dissipation=0, id="", manufacturer="", model="", price=0.0, fan_count=0, fan_speed_rpm=0):
        super().__init__(serial_number, installation_date, physically_installed, max_thermal_dissipation, id, manufacturer, model, price)
        self._fan_count = fan_count
        self._fan_speed_rpm = fan_speed_rpm

    # Setters
    def set_fan_count(self, fan_count):
        self._fan_count = fan_count

    def set_fan_speed_rpm(self, fan_speed_rpm):
        self._fan_speed_rpm = fan_speed_rpm

    # Getters
    def get_fan_count(self):
        return self._fan_count

    def get_fan_speed_rpm(self):
        return self._fan_speed_rpm

File: cpp/hardware/cooling/AirCooling.cpp
#pragma once
#include "Cooling.cpp"

class AirCooling : public Cooling
{
private:
    int fanCount;
    int fanSpeedRPM;

public:
    // Constructor
    AirCooling() : Cooling(), fanCount(0), fanSpeedRPM(0) {}
    AirCooling(const std::string &serialNumber, const Date &installationDate, bool physicallyInstalled, int thermalDissipation, const std::string &id, const std::string &manufacturer, const std::string &model, double price, int fanCount, int fanSpeedRPM)
        : Cooling(serialNumber, installationDate, physicallyInstalled, thermalDissipation, id, manufacturer, model, price), fanCount(fanCount), fanSpeedRPM(fanSpeedRPM) {}

    // Destructor
    ~AirCooling() {}

    // Setters
    void setFanCount(int fanCount)
    {
        this->fanCount = fanCount;
    }

    void setFanSpeedRPM(int fanSpeedRPM)
    {
        this->fanSpeedRPM = fanSpeedRPM;
    }

    // Getters
    int getFanCount() const
    {
        return fanCount;
    }

    int getFanSpeedRPM() const
    {
        return fanSpeedRPM;
    }
};